src/v/cloud_storage/partition_manifest.cc

```diff
@@ -789,6 +818,8 @@ struct partition_manifest_handler
                 _last_uploaded_compacted_offset = model::offset(u);
             } else if ("insync_offset" == _manifest_key) {
                 _insync_offset = model::offset(u);
+            } else if ("cloud_log_size_bytes" == _manifest_key) {
```

Comment:
we should keep in mind that this new manifest won't be readable by prev Redpanda (including RR)

Reply:
Do you mean that previous Redpanda versions won't be able to deserialise the manifest? I thought the new key just gets ignored.

Reply:
In this case handler will return an error and the rapidjson will throw an exception.

Reply:
It's tricky and I'm not sure it's a good thing, but you can add a new ambiguous state to the STM. If the handler sees an unexpected fields it should transition to this state, and when in this state it should just drop a value. Actually, there should be two ambiguous states, one for manifest itself and one for `segment_meta`.